MapzenSDK/MZMapViewController.swift

```diff
@@ -487,8 +487,10 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
    - parameter marker: The marker to add to map.
    */
   open func addMarker(_ marker: GenericMarker) {
-    currentMarkers[marker.tgMarker] = marker
-    marker.tgMarker.map = tgViewController
+    if marker.tgMarker == nil {
+      marker.tgMarker = tgViewController.markerAdd()
+    }
+    currentMarkers[marker.tgMarker!] = marker
```

Comment:
Ahhh slow github is slow - yeah the user data object is a good idea. Lemme think on how that might change the interaction model at the Mapzen layer.